tests/integration_tests/llms/test_azureml_endpoint.py

```diff
@@ -0,0 +1,137 @@
+    assert isinstance(output, str)
+
+
+def test_custom_formatter() -> None:
+    """Test ability to create a custom content formatter."""
+
+    class CustomFormatter(ContentFormatterBase):
+        content_type = "application/json"
+        accepts = "application/json"
+
+        def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+            input_str = json.dumps(
+                {
+                    "inputs": [prompt],
+                    "parameters": model_kwargs,
+                    "options": {"use_cache": False, "wait_for_model": True},
+                }
+            )
+            return input_str.encode("utf-8")
+
+        def format_response_payload(self, output: bytes) -> str:
+            response_json = json.loads(output)
+            return response_json[0]["summary_text"]
+
+    llm = AzureMLOnlineEndpoint(
+        endpoint_api_key=os.getenv("BART_ENDPOINT_API_KEY"),
+        endpoint_url=os.getenv("BART_ENDPOINT_URL"),
+        deployment_name=os.getenv("BART_DEPLOYMENT_NAME"),
+        content_formatter=CustomFormatter(),
+    )
+    output = llm("Foo")
+    assert isinstance(output, str)
+
+
+def test_missing_content_formatter() -> None:
+    """Test AzureML LLM without a body_handler attribute"""
+    with pytest.raises(AttributeError):
+        llm = AzureMLOnlineEndpoint(
+            endpoint_api_key=os.getenv("OSS_ENDPOINT_API_KEY"),
+            endpoint_url=os.getenv("OSS_ENDPOINT_URL"),
+            deployment_name=os.getenv("OSS_DEPLOYMENT_NAME"),
+        )
+        llm("Foo")
+
+
+def test_invalid_request_format() -> None:
+    """Test invalid request format."""
+
+    class CustomContentFormatter(ContentFormatterBase):
+        content_type = "application/json"
+        accepts = "application/json"
+
+        def format_request_payload(self, prompt: str, model_kwargs: Dict) -> bytes:
+            input_str = json.dumps(
+                {
+                    "incorrect_input": {"input_string": [prompt]},
+                    "parameters": model_kwargs,
+                }
+            )
+            return str.encode(input_str)
+
+        def format_response_payload(self, output: bytes) -> str:
+            response_json = json.loads(output)
+            return response_json[0]["0"]
+
+    with pytest.raises(json.JSONDecodeError):
+        llm = AzureMLOnlineEndpoint(
+            endpoint_api_key=os.getenv("OSS_ENDPOINT_API_KEY"),
+            endpoint_url=os.getenv("OSS_ENDPOINT_URL"),
+            deployment_name=os.getenv("OSS_DEPLOYMENT_NAME"),
+            content_formatter=CustomContentFormatter(),
+        )
+        llm("Foo")
+
+
+def test_saving_loading_llm(tmp_path: Path) -> None:
+    """Test saving/loading an AzureML Foundation Model LLM."""
+
+    llm = AzureMLOnlineEndpoint(
+        model_kwargs={"temperature": 0.03, "top_p": 0.4, "max_tokens": 200}
+    )
+    llm.save(file_path=tmp_path / "azureml.yaml")
+    loaded_llm = load_llm(tmp_path / "azureml.yaml")
```

Comment:
Can you actually make a call from loaded_llm in this test and assert the result.